main/src/cgeo/geocaching/filters/core/GeocacheFilterContext.java

```diff
@@ -14,14 +14,19 @@ public class GeocacheFilterContext implements Parcelable {
     public enum FilterType {
         LIVE(R.string.cache_filter_contexttype_live_title),
         OFFLINE(R.string.cache_filter_contexttype_offline_title),
-        TRANSIENT(R.string.cache_filter_contexttype_transient_title);
+        TRANSIENT(R.string.cache_filter_contexttype_transient_title),
+        NESTED(R.string.cache_filter_contexttype_nestedfilter_title);
```

Comment:
You are abusing the "FilterType" to transport an information inside nested GeocacheFilterActivities (whether the activity was opened in context of a nested filter or a "main" filter). This is not good, FilterType serves another purpose which is to globally allocate different filter storages. Please remove "NESTED" and use an own dedicated extra-parameter instead (e.g. a boolean parameter "isNested" or something like that)

Reply:
Will do...